gremsy_base/src/ros_gremsy.cpp

```diff
@@ -81,119 +73,39 @@ GimbalNode::GimbalNode(ros::NodeHandle nh, ros::NodeHandle pnh)
 
 void GimbalNode::gimbalStateTimerCallback(const ros::TimerEvent& event)
 {
-    // Publish Gimbal IMU
-    mavlink_raw_imu_t imu_mav = gimbal_interface_->get_gimbal_raw_imu();
-    imu_mav.time_usec = gimbal_interface_->get_gimbal_time_stamps().raw_imu; // TODO implement rostime
-    sensor_msgs::Imu imu_ros_mag = convertImuMavlinkMessageToROSMessage(imu_mav);
-    imu_pub.publish(imu_ros_mag);
-
     // Publish Gimbal Encoder Values
-    mavlink_mount_status_t mount_status = gimbal_interface_->get_gimbal_mount_status();
+    attitude<int16_t> encoder_values = gimbal_interface_->get_gimbal_encoder();
+
     geometry_msgs::Vector3Stamped encoder_ros_msg;
     encoder_ros_msg.header.stamp = ros::Time::now();
-    encoder_ros_msg.vector.x = ((float) mount_status.pointing_b) * DEG_TO_RAD;
-    encoder_ros_msg.vector.y = ((float) mount_status.pointing_a) * DEG_TO_RAD;
-    encoder_ros_msg.vector.z = ((float) mount_status.pointing_c) * DEG_TO_RAD;
-    // encoder_ros_msg.header TODO time stamps
-
-    encoder_pub.publish(encoder_ros_msg);
-
-    // Get Mount Orientation
-    mavlink_mount_orientation_t mount_orientation = gimbal_interface_->get_gimbal_mount_orientation();
-
-    yaw_difference_ = DEG_TO_RAD * (mount_orientation.yaw_absolute - mount_orientation.yaw);
-
-    // Publish Camera Mount Orientation in global frame (drifting)
-    mount_orientation_incl_global_yaw.publish(
-        tf2::toMsg(
-            convertYXZtoQuaternion(
-                mount_orientation.roll,
-                mount_orientation.pitch,
-                mount_orientation.yaw_absolute)));
-
-    // Publish Camera Mount Orientation in local frame (yaw relative to vehicle)
-    mount_orientation_incl_local_yaw.publish(
-        tf2::toMsg(
-            convertYXZtoQuaternion(
-                mount_orientation.roll,
-                mount_orientation.pitch,
-                mount_orientation.yaw)));
-}
+    encoder_ros_msg.vector.x = ((float) encoder_values.roll);
```

Comment:
Is the SDK interface changed to radians now?

Reply:
It works directly with degrees. There would be no problem to change it to radians if preferred.